src/domain/core/src/utils/time_source.rs

```diff
@@ -24,46 +28,132 @@ pub trait SystemTimeSource: Send + Sync {
 #[dill::interface(dyn SystemTimeSource)]
 pub struct SystemTimeSourceDefault;
 
+#[async_trait::async_trait]
 impl SystemTimeSource for SystemTimeSourceDefault {
     fn now(&self) -> DateTime<Utc> {
         Utc::now()
     }
+
+    async fn wake_after(&self, duration: Duration) {
```

Comment:
why not call it `sleep`?

Reply:
In general, yes, this is the most expected name for this method -- I chose the current name under impression of the ticket requirements.

@zaychenko-sergei , please share your thoughts -- or do I need to just use `sleep()`?

Reply:
`advance()` was chosen
based on https://github.com/kamu-data/kamu-cli/pull/451#discussion_r1459404953

Reply:
Name is a least of interesting problems here. I'm okay with any of these suggestions.

Reply:
renamed to `sleep()`